src/DynamoCoreWpf/Views/PackageManager/Controls/SearchBoxControl.xaml.cs

```diff
@@ -16,28 +14,28 @@ namespace Dynamo.PackageManager.UI
     public partial class SearchBoxControl : UserControl
     {
         private DispatcherTimer delayTimer;
+        private string userSearchInput = ""; // Store user's input separately
+        private static int delayTime = 100;  // set delay for event 100ms
```

Comment:
Isn't there already a "Delay" property on a Binding object ?
ex: Value="{Binding ElementName=ValueText, Delay=100, Path=Text, Mode=TwoWay}"
Wouldn't that work ?

Reply:
God bless you @pinzart90 ! I did not know this property existed 😶‍🌫️ 

Thank you so much! This cleaned up so much complexity and code!!